dev/release/verify-release-candidate.sh

```diff
@@ -904,6 +904,7 @@ test_integration() {
   maybe_setup_virtualenv
 
   pip install -e dev/archery[integration]
+  pip install jpype1
```

Comment:
nit: this should be added as an archery dep?

Reply:
I am currently confused on what to do to be honest because I asked the same thing on the issue :) https://github.com/apache/arrow/issues/38470#issuecomment-1780956462

Reply:
IMHO, no, because this is only for testing the Java side of the C Data Interface.

Reply:
Can we create a separated configuration something like `integration-java` and use it like `pip install -e dev/archery[integration-java]`?